### ISSUE TITLE
[CRITICAL] Code Review: 6 funktionale Bugs — Resume-Verlust, DataStore-Crash, Silent Data Loss

### ISSUE THREAD

**Issue by karlokarate** [karlokarate/FishIT-Player#759]
## Code Review — CRITICAL Findings

Full codebase review (alle v2 Layer). Diese 6 Bugs verursachen **Datenverlust, Crashes oder grundlegend kaputte Funktionalität**.

---

### C1: `release()` speichert Resume-Position nicht

**File:** `player/internal/src/main/java/com/fishit/player/internal/session/InternalPlayerSession.kt` (L418)

`release()` ruft `saveResumePosition()` **nicht** auf bevor der Player freigegeben wird. Resume-Daten gehen verloren bei:
- App-Wechsel / Hintergrund
- Stream-Wechsel via `switchStream()` (ruft intern `release()` auf)

```kotlin
fun release() {
    positionUpdateJob?.cancel()
    subtitleTrackManager.detach()
    audioTrackManager.detach()
    player?.removeListener(playerListener)
    player?.release()  // ← Resume-Position weg
    player = null
    _state.value = InternalPlayerState.INITIAL
}
```

**Fix:** `saveResumePosition()` vor `player?.release()` aufrufen.

---

### C2: `destroy()` kann laufende Resume-Speicherung abbrechen

**File:** `player/internal/src/main/java/com/fishit/player/internal/session/InternalPlayerSession.kt` (L429)

```kotlin
fun destroy() {
    release()          // ← State → INITIAL
    scope.cancel()     // ← Kills in-flight saveResumePosition() via CancellationException
}
```

**Fix:** `saveResumePosition()` mit `NonCancellable` context vor `release()` aufrufen.

---

### C3: Dual `SyncCheckpointStore` — DataStore Runtime Crash

**Files:**
- `core/catalog-sync/…/SyncCheckpointStore.kt` — source-spezifisch (`getXtreamCheckpoint()`)
- `core/sync-common/…/SyncCheckpointStore.kt` — generisch (`getCheckpoint(key)`)

Beide deklarieren `preferencesDataStore("sync_checkpoints")` → **zwei DataStore-Delegates mit gleichem Dateinamen** → conflicting file locks → **RuntimeException** auf Android.

`XtreamCatalogSync` injected die generische Version, `DefaultTelegramSyncService` die source-spezifische.

**Fix:** Zu EINER Implementierung konsolidieren (SSOT-Verletzung).

---

### C4: Live-Suche — Favorites immer leer

**File:** `infra/data-nx/…/live/NxLiveContentRepositoryImpl.kt` (L117)

```kotlin
val favoriteWorkKeys = userStateRepository
    .observeFavorites(DEFAULT_PROFILE_KEY, limit = FAVORITES_LIMIT)
    .let { emptySet<String>() }  // ← Flow-Ergebnis wird verworfen!
```

Gleicher Bug bei `getRecentChannels()` (L141): `continueWatching` ist immer eine leere Liste.

**Fix:** `.first().toSet()` statt `.let { emptySet() }`.

---

### C5: `ContainerGuess` — M4A als MP4 fehlidentifiziert

**File:** `core/model/…/util/ContainerGuess.kt` (L17)

```kotlin
"mp4" in mimeType -> "mp4"   // ← Fängt auch "audio/mp4" ab!
// ...
"m4a" in mimeType || ... -> "m4a"  // ← Unerreichbar für audio/mp4
```

`audio/mp4` ist der Standard-IANA-Typ für M4A (AAC-in-MP4). Wird fälschlich als `"mp4"` zurückgegeben.

**Fix:** Audio-MIME-Checks VOR Video prüfen, oder `&& "audio" !in mimeType` in der ersten Branch.

---

### C6: `ImageRef.parsePrefixedFormat` — HTTP URL-Schema fehlt

**File:** `core/model/…/ImageRef.kt` (L220)

```kotlin
"http" -> Http(url = value)           // ← value hat KEIN Schema-Prefix → kaputte URL
"https" -> Http(url = "https:$value") // ← korrekt rekonstruiert
```

Inkonsistente URL-Rekonstruktion: `http`-Case liefert URL ohne Schema.

**Fix:** `"http"` Case auf `Http(url = "http:$value")` ändern (analog zu `https`).

---

## Checkliste

- [ ] C1: `saveResumePosition()` in `release()` aufrufen
- [ ] C2: `saveResumePosition()` mit `NonCancellable` in `destroy()`
- [ ] C3: `SyncCheckpointStore` konsolidieren (SSOT)
- [ ] C4: Live favorites/recent channels Fix
- [ ] C5: ContainerGuess audio/mp4 → m4a
- [ ] C6: ImageRef http Schema-Prefix

**Comment by karlokarate** [comment]
## Agent A — Final Report (core:model)

### Compile Status: ✅ BUILD SUCCESSFUL
`./gradlew :core:model:compileDebugKotlin` — 10 tasks, 0 errors.

---

### All Fixed Findings

| ID | Sev | Fix | File |
|----|-----|-----|------|
| **C5** | CRIT | Moved audio mp4 checks (`m4a`, `mp4a`, `audio/mp4`) **before** generic `mp4` branch. Removed dead-code `"mp4" !in mimeType` guard. Now: `audio/mp4` → `"m4a"`, `audio/mp4a-latm` → `"m4a"`, `video/mp4` → `"mp4"`. | `core/model/.../util/ContainerGuess.kt` |
| **C6** | CRIT | `"http"` case now returns `Http(url = "http:$value")` — symmetric with `"https"` case. Previously dropped the scheme prefix. | `core/model/.../ImageRef.kt` L221 |
| **H15** | HIGH | KDoc on `addedTimestamp` corrected from "seconds" to **"milliseconds"** in both `RawMediaMetadata` and `NormalizedMediaMetadata`. Added unit documentation and pipeline contract note ("Pipelines MUST convert seconds-based source values to millis before assigning"). Both fields are now documented as `Long` millis consistent with `System.currentTimeMillis()`. | `RawMediaMetadata.kt`, `NormalizedMediaMetadata.kt` |
| **M11** | MED | Resolution score normalized: `(height / 360).coerceAtMost(10)` — max 6 for 4K vs max 40 for language. Language preference now dominates as originally documented in KDoc. | `VariantSelector.kt` L103-104 |
| **M12** | MED | `NormalizedMedia.variants` changed from `MutableList<MediaVariant>` to `List<MediaVariant>`. Prevents shared-reference mutation after `copy()`. | `NormalizedMedia.kt` L79 |
| **M13** | MED | `MediaVariant.available` changed from `var` to `val`. Restores hash-stability for hash-based collections. | `MediaVariant.kt` L27 |
| **S3** | MED | New `object DurationFormatter` created at `core/model/util/DurationFormatter.kt` as SSOT with `formatFriendly(ms)` and `formatPrecise(ms)`. `MediaSourceRef` now delegates to it. Added to SSOT registry in `ssot-enforcement.instructions.md`. | `DurationFormatter.kt` (new), `MediaSourceRef.kt` |

---

### ⚠️ BREAKING PUBLIC API CHANGES — Agents B–E must adapt callers

#### 1. `NormalizedMedia.variants: List<MediaVariant>` (was `MutableList`)

| Affected Caller | Module | Required Change |
|-----------------|--------|-----------------|
| `Normalizer.kt` L75-81 | `core/metadata-normalizer` | Replace `variants.clear()` + `variants.addAll(sorted)` with direct list construction. Pass `listOf(...)` or sorted result to constructor instead of `mutableListOf()`. |
| `Normalizer.kt` L109 | `core/metadata-normalizer` | Replace `mutableListOf(variant)` with `listOf(variant)` |

#### 2. `MediaVariant.available: val` (was `var`)

| Affected Caller | Module | Required Change |
|-----------------|--------|-----------------|
| `VariantPlaybackOrchestrator.kt` L120 | `player/internal` | Replace `variant.available = false` with `variant.copy(available = false)` and rebuild the variants list in the parent `NormalizedMedia`. |

#### 3. `RawMediaMetadata.addedTimestamp` — KDoc now states **milliseconds**

| Affected Caller | Module | Required Change |
|-----------------|--------|-----------------|
| `XtreamSyncStatusFactory.kt` L138 | `core/catalog-sync` | **ACTIVE BUG:** Remove `* 1000L` — value is already millis. Currently: `(raw.addedTimestamp ?: 0L) * 1000L` produces year ~57000 timestamps → incremental sync never skips items. Fix: `raw.addedTimestamp ?: 0L` |

#### 4. `DurationFormatter` SSOT — Duplicates to consolidate

| Duplicate Location | Module | Required Change |
|--------------------|--------|-----------------|
| `DetailResumeInfo.kt` L88 | `feature/detail` | Delete private `formatDuration()`, use `DurationFormatter.formatPrecise()` |
| `SourceBadge.kt` L603 | `feature/detail` | Delete private `formatDuration()`, use `DurationFormatter.formatPrecise()` |

---

### Chain Position
- **Predecessor:** None — Agent A starts the chain
- **Successor:** Agent B (`core/persistence`)
- **Status:** ✅ COMPLETE — all 7 findings fixed, module compiles, report posted
